homeassistant/components/homematicip_cloud/config_flow.py

```diff
@@ -101,8 +104,9 @@ def __init__(self) -> None:
         if hapid in configured_haps(self.hass):
             return self.async_abort(reason="already_configured")
 
-        _LOGGER.info("Imported authentication for %s", hapid)
+        await self.async_set_unique_id(hapid)
```

Comment:
Set this as early as possible and then call `self._abort_if_unique_id_configured`.

https://github.com/home-assistant/home-assistant/blob/f4626375f3e5a44d62fc08f2b9e833e456c61147/homeassistant/config_entries.py#L819-L820

Then we can remove the above check for configured haps.

Reply:
Done